lib/broccoli/ember-app.js

```diff
@@ -1845,6 +1845,30 @@ function addOutputFile(container, assetPath, options) {
     container[outputFile] = [];
   }
 
+  if (~container[outputFile].indexOf(assetPath)) {
+    var message = [
+      'You have specified a duplicate asset via `app.import(' + assetPath + ')`.',
+      'This can result in very strange behavior in your application as the',
+      'asset historically was included twice. Multiple versions of the same asset',
+      'are also clobbered by the last version added to the Broccoli tree, meaning',
+      'that it only ever included the version of the asset which was last added',
+      'to the Broccoli tree. This should be treated as a bug to address. This',
+      'typically occurs when an asset is imported by a nested addon. Ember CLI',
+      'no longer includes multiple copies by default and instead is using only',
+      'the _last_ occurrence imported.'
+    ];
+    this.project.ui.writeWarnLine(message.join(' '));
```

Comment:
> The "right thing" here is nebulous.

Yup, I intended to leave "right thing" vague, so we could pencil it in as materializes clearly.

> Dedupe strategy is already defined and tested: last import wins (by array order, not invocation order because prepend).

Sounds good